### PR TITLE
feat: moving markdown editing to monaco

### PR DESCRIPTION
**Pull request by drdelambre**
this moves the markdown editor to monaco so that we can have a more unified "editing code" experience.

has autogrow, theming, all that great stuff

### CODE REVIEW

File: ui/src/external/markdown.tmLanguage.json [hunk 1]
```diff
@@ -0,0 +1,2628 @@
...
+                    "name": "fenced_code.block.language.attributes.markdown"
+                }
+            },
+            "end": "(^|\\G)(\\2|\\s{0,3})(\\3)\\s*$",
+            "endCaptures": {
+                "3": {
+                    "name": "punctuation.definition.markdown"
+                }
+            },
+            "name": "markup.fenced_code.block.markdown",
```
Comment (drdelambre) [inline ui/src/external/markdown.tmLanguage.json:352]: its been awhile, but i'm pretty sure you find names here.....

File: ui/src/external/monaco.markdown.theme.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    rules: [],
+    colors: {
+      'editor.foreground': '#F8F8F8',
+      'editor.background': '#202028',
+      'editorGutter.background': '#25252e',
+      'editor.selectionBackground': '#353640',
+      'editorLineNumber.foreground': '#666978',
+      'editor.lineHighlightBackground': '#353640',
+      'editorCursor.foreground': '#ffffff',
+      'editorActiveLineNumber.foreground': '#bec2cc',
```
Comment (alexpaxton) [inline ui/src/external/monaco.markdown.theme.ts:18]: Nais I'll have to play around with this

File: ui/src/external/monaco.markdown.theme.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import {MonacoType} from 'src/types'
+
+const THEME_NAME = 'markdownTheme'
+
+function addTheme(monaco: MonacoType) {
+  monaco.editor.defineTheme(THEME_NAME, {
+    base: 'vs-dark',
+    inherit: false,
+    rules: [],
```
Comment (drdelambre) [inline ui/src/external/monaco.markdown.theme.ts:9]: ...and add them to the rules here. take a look at the monaco.toml.theme.ts and monaco.flux.theme.ts in the same folder

File: ui/src/notebooks/pipes/markdown/index.ts [hunk 4]
```diff
@@ -4,12 +4,173 @@ import './style.scss'
 
 export type MarkdownMode = 'edit' | 'preview'
 
+const TEST_MODE = false
```
Comment (alexpaxton) [inline ui/src/notebooks/pipes/markdown/index.ts:7]: Nice

**Pull request merged**